phoenicis-repository/src/main/java/org/phoenicis/repository/repositoryTypes/LocalRepository.java

```diff
@@ -101,8 +101,15 @@ public RepositoryDTO fetchInstallableApplications() {
 
                 if (categoryJson.exists()) {
                     final CategoryDTO.Builder categoryDTOBuilder = new CategoryDTO.Builder(
-                            unSerializeCategory(categoryJson)).withId(categoryDirectory.getName())
-                                    .withApplications(fetchApplications(categoryDirectory));
+                            unSerializeCategory(categoryJson)).withApplications(fetchApplications(categoryDirectory));
+
+                    if (StringUtils.isBlank(categoryDTOBuilder.getId())) {
```

Comment:
That would mean that
1. the category.json contains an ID (which is not the case currently)
2. this ID is invalid

But yes, in that case we might get into trouble.